executor/runtime/docker/docker.go

```diff
@@ -2108,7 +2109,7 @@ func (r *DockerRuntime) setupGPU(ctx context.Context) error {
 }
 
 // Kill uses the Docker API to terminate a container and notifies the VPC driver to tear down its networking
-func (r *DockerRuntime) Kill(ctx context.Context) error { // nolint: gocyclo
+func (r *DockerRuntime) Kill(ctx context.Context, wasKilled bool) error { // nolint: gocyclo
 	logger.G(ctx).Debug("Shutting down main container")
```

Comment:
Can you do:
```
if wasKilled {
  logger.G(ctx).Debug("Shutting down main container because we were asked to stop from the API")
} else {
  logger.G(ctx).Debug("Shutting down main container because it finished or died")
}
```
There have been a number of times where I which I had that distinction in the logs, and thanks to you, now we can know

Reply:
Good idea - will do.